keras_nlp/models/xlnet/xlnet_backbone.py

```diff
@@ -0,0 +1,217 @@
+    model = keras_nlp.models.XLNetBackbone(
+        vocabulary_size=32000,
+        num_layers=12,
+        num_heads=12,
+        hidden_dim=768,
+        intermediate_dim=3072,
+    )
+    output = model(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        vocabulary_size,
+        num_layers,
+        num_heads,
+        hidden_dim,
+        intermediate_dim,
+        dropout=0.0,
+        activation="gelu",
+        kernel_initializer_range=0.02,
+        bias_initializer="zeros",
+        layer_norm_epsilon=1e-12,
+        **kwargs,
+    ):
+        # Inputs
+        token_id_input = keras.Input(
+            shape=(None,), dtype="int32", name="token_ids"
+        )
+        padding_mask = keras.Input(
+            shape=(None,), dtype="int32", name="padding_mask"
+        )
+        segment_ids = keras.Input(
+            shape=(None,), dtype="int32", name="segment_ids"
+        )
+
+        # Content and Query Embedding
+        word_emb, pos_emb = ContentAndQueryEmbedding(
+            vocabulary_size=vocabulary_size,
+            hidden_dim=hidden_dim,
+            dropout=dropout,
+            name="content_query_embedding",
+        )(token_id_input=token_id_input)
+
+        # Apply XLNetAttentionMaskLayer and XLNetSegmentMatrixLayer Layers
+        # to get the processed attention masks and segment matrix.
+        attn_mask_content, attn_mask_query = XLNetAttentionMaskLayer(
+            hidden_dim=hidden_dim,
+            kernel_initializer_range=kernel_initializer_range,
+            name="encoder_block_attn_mask_layer",
+        )(padding_mask)
+        seg_mat = XLNetSegmentMatrixLayer(name="encoder_block_seg_mat_layer")(
+            segment_ids
+        )
+
+        output_content = word_emb
+
+        # Encoders
+        head_dim = hidden_dim // num_heads
+        for i in range(num_layers):
+            output_content, output_query = XLNetEncoder(
+                num_heads=num_heads,
+                hidden_dim=hidden_dim,
+                head_dim=head_dim,
+                intermediate_dim=intermediate_dim,
+                dropout=dropout,
+                activation=activation,
+                layer_norm_epsilon=layer_norm_epsilon,
+                kernel_initializer_range=kernel_initializer_range,
+                bias_initializer=bias_initializer,
+                name=f"xlnet_encoder_{i}",
+            )(
+                output_content=output_content,
+                attn_mask_content=attn_mask_content,
+                attn_mask_query=attn_mask_query,
+                pos_emb=pos_emb,
+                seg_mat=seg_mat,
+            )
+
+        output = keras.layers.Dropout(dropout)(output_content)
+
+        super().__init__(
+            inputs={
+                "token_ids": token_id_input,
+                "padding_mask": padding_mask,
+                "segment_ids": segment_ids,
+            },
+            outputs={"last_hidden_state": output},
```

Comment:
When only one output, omit the dictionary.

Reply:
ok